controlplane/kubeadm/internal/workload_cluster.go

```diff
@@ -545,3 +555,28 @@ func yamlToUnstructured(rawYAML []byte) (*unstructured.Unstructured, error) {
 	err := yaml.Unmarshal(rawYAML, unst)
 	return unst, err
 }
+
+// ImageRepositoryFromClusterConfig returns the image repository to use. It returns:
+// * clusterConfig.ImageRepository if set.
+// * "registry.k8s.io" if v1.25 <= version < v1.26 to migrate to the new registry
+// * "" otherwise.
+// Beginning with kubernetes v1.25, kubernetes images get published to registry.k8s.io
```

Comment:
Please update the comment as well. As mentioned above they are available under both